apache-airflow/docker-compose.yml

```diff
@@ -1,9 +1,18 @@
 version: '3.4'
 
 services:
+  db:
+    image: mysql
+    build: ./mysql
+    command: --default-authentication-plugin=mysql_native_password
+    environment:
+      MYSQL_ROOT_PASSWORD: password
+
   notebook:
     image: mattjtodd/docker-airflow
-    build: ./
+    links:
+     - db
```

Comment:
Fair enough. At that point (failing to find another way to just test such connection - ping/telnet, etc) I just added it just in case. It is gone now - but I'm not going to test it today...

Reply:
You should be able to `ping db` from the notebook container console, unless the jupyter user doesn't haver permissions.  I've just checked this and is it because you don't have the ping application installed?  If this is the case you have two options.  

1. Install ping into the images via the Dockerfile's apt package management system, then try again. 
2. Create a new container which share's it's network namespace and has these tools installed.  This looks like:

`docker run --rm -it --net container:apache-airflow_notebook_1 alpine`

This will open a shell and give you access to test the networking namespace of the notebooks container.

I just tested this and the db service is resolvable, so you should be good to configure any uri for database access to use the dns name `db`.

Reply:
Hope that's helped clarify things, if not give me a shout, happy to explain.

Reply:
Oh, all worked 'on my machine' :)

I just mentioned it as ping/telnet/ifconfig/etc are not installed. Then I tried do it adhoc but user didn't have permissions and couldn't just do sudo... Dropping everything and recreating it with network utils installed was too much a hassle - so I just updated connection and hoped for the best (and it worked - :) )